SilKit/source/capi/Test_CapiExceptions.cpp

```diff
@@ -0,0 +1,178 @@
+/* Copyright (c) 2022 Vector Informatik GmbH
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. */
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "CapiImpl.hpp"
+#include "silkit/detail/impl/ThrowOnError.hpp"
+
+namespace {
+
+class Test_CapiExceptions : public testing::Test
+{
+public:
+    Test_CapiExceptions() {}
+};
+
+SilKit_ReturnCode throw_SilKitError()
```

Comment:
You could merge all of these into a single function template, like so:

```c++
template<typename T>
auto TestExceptionToErrorCode()
try
{
    throw T{"error msg"};
}
CAPI_CATCH_EXCEPTIONS
```

And invoke it like so:

```c++
EXPECT_EQ(TestExceptionToErrorCode<SilKit::LengthError>(), SilKit_ReturnCode_LENGTH_ERROR);
```